snapcraft/tests/fixture_setup.py

```diff
@@ -548,3 +548,50 @@ def setUp(self):
             revno = call_with_output(['hg', 'id']).split()[0]
 
             self.commit = revno
+
+
+class FakeAptCache(fixtures.Fixture):
```

Comment:
This is the most simple fake to get this working.
With some more time, and patience to read the apt source code we could use a fake /etc/apt directory instead. I think this is good for now.